### PR TITLE
DOCS-554-601: revise frame system page

### PR DESCRIPTION
**Pull request by sguequierre**
* updating structure of page
* moved reference section out to internals re. docs team discussion
* trying to remove duplicate or inaccurate information 
* made some of the trickier revisions with @spieswl guidance (like changing tree image from _offset to _origin, geometries information, orientation vectors type notes)

### CODE REVIEW

File: docs/internals/kinematic-chain-config.md [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+weight: 70
+type: "docs"
+description: "How to write files defining kinematic parameters to configure intermediate reference frames for components with complex kinematic chains."
+tags: ["slam", "services"]
+# SMEs: Motion
+---
+
+Many components have complex kinematic chains and require an additional set of intermediate reference frames to use Viam's [motion service](/services/motion/).
+
+- For example, an [arm](/components/arm) has a reference frame originating where the arm is attached to a surface, but it also has links and joints whose frames of reference matter when attempting to move the arm to a [pose](/internals/pose) with [`MoveToPositions()`](/components/arm/#movetopositions).
```
Comment (spieswl) [inline docs/internals/kinematic-chain-config.md:13]:
```suggestion
- For example, an [arm](/components/arm) has a reference frame originating where the arm is attached to a surface, but it also has links and joints whose frames of reference matter when attempting to move the arm to a [pose](/internals/pose) with [`MoveToPosition()`](/components/arm/#movetoposition).
```

File: docs/internals/kinematic-chain-config.md [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+
+If you want to implement a component with a complex kinematic chain that is not already built into the RDK to build your robot with Viam, you need to add a file to your driver that details the attachment of the intermediate reference frames on the component.
```
Comment (spieswl) [inline docs/internals/kinematic-chain-config.md:15]:
```suggestion
If you want to implement a component with a complex kinematic chain that is not already built into the RDK, you need to add a file to your driver that details the attachment of the intermediate reference frames on the component.
```

Hopefully, modular arms will be ready by the time people are doing this. At present, RDK modifications are needed to add an arbitrary component's physical description.

File: docs/services/frame-system/_index.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+type: docs
+weight: 45
+no_list: true
+tags: ["frame system", "services"]
+# SMEs: Peter L, Gautham, Bijan
+---
+
+Any robot configured in Viam comes with a service we call the Frame System: an internally managed and mostly static system for storing the "reference frame" of each component of a robot within a coordinate system configured by the user.
+
+The Frame System is the basis for many of Viam's other services, like [motion](/services/motion) and [vision](/services/vision).
```
Comment (spieswl) [inline docs/services/frame-system/_index.md:14]:
```suggestion
The Frame System is the basis for many of Viam's other services, like [Motion](/services/motion) and [Vision](/services/vision).
```

File: docs/services/frame-system/_index.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+{{% alert title="Caution" color="caution" %}}
+
+`Types` are offered in `Orientation` for ease of configuration, but the [orientation vector](/internals/orientation-vector/) type that is always stored and returned by the Frame System is `"Orientation Vector Radians"`.
+`"Degrees"` and `"Quaternion"` will be converted to `"Radians"`.
+
+{{% /alert %}}
+
+{{% alert title="Tip" color="tip" %}}
+
+Viam's coordinate system considers `+X` to be forward, `+Y` to be left, and `+Z` to be up.
```
Comment (spieswl) [inline docs/services/frame-system/_index.md:111]: [suggest] Make mention of the "right hand rule"? https://en.wikipedia.org/wiki/Right-hand_rule

File: docs/services/frame-system/_index.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+## Accessing the Frame System
+
+The [Robot API](https://github.com/viamrobotics/api/blob/main/proto/viam/robot/v1/robot.proto) supplies two methods to interact with the Frame System through gRPC calls:
+
+1. `TransformPose`: Transforms a pose measured in one reference frame to the same pose as it would have been measured in another.
+2. `FrameSystemConfig`: Returns a topologically sorted list of all the reference frames monitored by the frame system.
+Any [supplemental transforms](#handling-motion-with-supplemental-transforms) are also merged into the tree, topologically sorted, and returned.
+
+## Supplemental Transforms
```
Comment (spieswl) [inline docs/services/frame-system/_index.md:149]: I need to read over this section more thoroughly, there is a decent amount of information here that feels outdated and I want to double-check that.

**Review by spieswl — CHANGES_REQUESTED** [review]
Marking as "Request Changes" for a couple suggestions, and until I've had a chance to review with Peter, Bijan, and Gautham some of the original ideas here.

File: docs/services/frame-system/_index.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+| Parameter | Inclusion | Required |
+| --------- | ----------- | ----- |
+| `Parent`  | **Required** | Default: `world`. The name of the reference frame you want to act as the parent of this frame. |
+| `Translation` | **Required** | Default: `(0, 0, 0)`. The coordinates that the origin of this component's reference frame has within its parent reference frame. |
+| `Orientation`  | **Required** | Default: `(0, 0, 1), 0`. The [orientation vector](/internals/orientation-vector/) that yields the axes of the component's reference frame when applied as a rotation to the axes of the parent reference frame. <br> Types: `Orientation Vector Degrees`, `Orientation Vector Radians`, and `Quaternion`. |
+| `Geometry`  | **Required** | Default: `none`. Collision geometries for defining bounds in the environment of the robot. <br> Types: `Sphere`, `Box`, and `Capsule`. |
+
+{{% alert title="Caution" color="caution" %}}
+
+`Types` are offered in `Orientation` for ease of configuration, but the [orientation vector](/internals/orientation-vector/) type that is always stored and returned by the Frame System is `"Orientation Vector Radians"`.
```
[the remark this answers is not in the document]
Comment (npentrel) [reply docs/services/frame-system/_index.md:104]: can we put this caution alert in the table?

File: docs/services/frame-system/_index.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+For more information, see these two examples:
+
+- [A component attached to a static surface](/component-on-static)
+- [A component attached to another, dynamic, component](/component-on-dynamic)
+
+## Building the Frame System
+
+Once you configure your robot and run `viam-server`, your robot builds a tree of reference frames with the `world` as the root node.
+
+A [topologically-sorted list](https://en.wikipedia.org/wiki/Topological_sorting) of the generated reference frames is printed by the server and can be seen in the logs at `--debug` level:
```
Comment (npentrel) [inline docs/services/frame-system/_index.md:124]: remove passive voice

File: docs/services/frame-system/_index.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+`viam-server` builds the connections in this tree by looking at the `"frame"` portion of each component in the robot's configuration and defining *two* reference frames for each component:
+
+1. One with the name of the component, representing the actuator or final link in the component's kinematic chain: like `"A"` as the end of an arm.
+2. Another representing the origin of the component, defined with the component's name and the suffix *"_origin"*.
+
+## Accessing the Frame System
+
+The [Robot API](https://github.com/viamrobotics/api/blob/main/proto/viam/robot/v1/robot.proto) supplies two methods to interact with the Frame System through gRPC calls:
+
+1. `TransformPose`: Transforms a pose measured in one reference frame to the same pose as it would have been measured in another.
```
Comment (npentrel) [inline docs/services/frame-system/_index.md:145]: We should probably have the api methods here like in components

File: docs/services/frame-system/_index.md [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+
+However, a dynamic arm attached to a [dynamic rover](/components/base/wheeled) could not be configured into the Frame System if the rover is unaware of its own position, because the rover can move freely with respect to the `world` reference frame.
+
+To solve this problem:
+
+- You can introduce a [movement sensor](/components/movement-sensor) or a [camera](/components/camera), in combination with our [Vision Service](/services/vision/), as a third component.
+- This component would be fixed in respect to the `world` reference frame, and could supply the location and orientation of the rover in its own reference frame.
+
+To add this component's reference frame into your robot's Frame System build, you can pass the `name` of this component's reference frame to various APIs.
+These *supplemental transforms* would be the missing link to be able to transform a [pose](/internals/orientation-vector) in that dynamic arm's reference frame to the `world` reference frame.
```
Comment (npentrel) [inline docs/services/frame-system/_index.md:163]:
```suggestion
These *supplemental transforms* then supply the missing link to transform a [pose](/internals/orientation-vector) in that dynamic arm's reference frame to the `world` reference frame.
```

File: docs/services/frame-system/_index.md [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+
+- For example, you can pass this component's reference frame information to the `supplemental_transforms` parameter in your calls to Viam's motion service [`GetPose`](/services/motion/#getpose) method.
+- Functions of some services and components also take in a `WorldState` parameter, which includes a `transforms` parameter.
+- Both [`TransformPose`](#accessing-the-frame-system) and [`FrameSystemConfig`](#accessing-the-frame-system) have the option to take in these supplemental transforms.
+
+{{% alert title="Tip" color="tip" %}}
+
+A knowledgeable user could code a [mobile base](/components/base/wheeled) that is able to report its own position without the need for supplemental transforms with an organic [SLAM](/services/slam) system.
```
Comment (npentrel) [inline docs/services/frame-system/_index.md:171]: Avoid conditional
```suggestion
Experiences users can code a [mobile base](/components/base/wheeled) that is able to report its own position without the need for supplemental transforms with an organic [SLAM](/services/slam) system.
```

**Review by npentrel — CHANGES_REQUESTED** [review]
A few comments - remind me what is happening with the Model JSON?

File: docs/internals/kinematic-chain-config.md
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/internals/kinematic-chain-config.md:1]: @npentrel This is the "Model JSON" section, in our sync we settled on moving it here but I definitely think there could be a better place for it-- motion team doesn't think it should be with frame system as it's not specific to that service. 

Re. changing title and not centering around "Model JSON", referring to this file format as that was confusing with the other uses of Model across Viam docs, and it made it seem like the file needed to be named "Model.json" or similar when that was not the case. Hoping that this document will do a better job of clearing up the purpose and use of these files

File: docs/internals/kinematic-chain-config.md [hunk 11]
```diff
@@ -0,0 +1,318 @@
+---
+title: "Configuring Complex Kinematic Chains"
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/internals/kinematic-chain-config.md:2]: done!

File: docs/internals/kinematic-chain-config.md (hunk 1 above)
Comment (sguequierre) [reply docs/internals/kinematic-chain-config.md:13]: Ah good catch, done!

File: docs/internals/kinematic-chain-config.md (hunk 2 above)
Comment (sguequierre) [reply docs/internals/kinematic-chain-config.md:15]: 👀 👍  done!

File: docs/internals/kinematic-chain-config.md [hunk 12]
```diff
@@ -0,0 +1,318 @@
...
+{{% alert title="Note" color="note" %}}
+
+Of the two methods, Viam prefers Spacial Vector Algebra over Denavit-Hartenberg.
+
+Viam wants roboticists to be able to specify link frames arbitrarily, which DH parameters are unable to guarantee.
+We also want roboticists to be able to make their own robots and define new drivers, and incorrect SVA parameters are much easier to troubleshoot than incorrect DH parameters.
+
+{{% /alert %}}
+
+Reference this example <file>.json</file> configuration for each `kinematic_param_type`, as used by Viam's [Universal Robots](https://www.universal-robots.com/) [arm driver](https://github.com/viamrobotics/rdk/blob/main/components/arm/universalrobots/ur5e.json):
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/internals/kinematic-chain-config.md:35]: done! Do you think it's still clear that these tabs vary by `kinematic_param_type`?

File: docs/services/frame-system/_index.md [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+linkTitle: "Frame System"
+description: "The Frame System holds reference frame information for the relative position of components in space."
+type: docs
+weight: 45
+no_list: true
+tags: ["frame system", "services"]
+# SMEs: Peter L, Gautham, Bijan
+---
+
+Any robot configured in Viam comes with a service we call the Frame System: an internally managed and mostly static system for storing the "reference frame" of each component of a robot within a coordinate system configured by the user.
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/services/frame-system/_index.md:12]: Ah yes, done!

File: docs/services/frame-system/_index.md (hunk 9 above)
Comment (sguequierre) [reply docs/services/frame-system/_index.md:163]: done!

File: docs/services/frame-system/_index.md (hunk 7 above)
Comment (sguequierre) [reply docs/services/frame-system/_index.md:124]: done!

File: docs/services/frame-system/_index.md (hunk 3 above)
Comment (sguequierre) [reply docs/services/frame-system/_index.md:14]: Done!

File: docs/internals/kinematic-chain-config.md [hunk 14]
```diff
@@ -0,0 +1,316 @@
...
+title: "Configure Complex Kinematic Chains"
+linkTitle: "Complex Kinematic Chains"
+weight: 70
+type: "docs"
+description: "How to write files defining kinematic parameters to configure intermediate reference frames for components with complex kinematic chains."
+tags: ["slam", "services"]
+# SMEs: Motion
+---
+
+Many components have complex kinematic chains and require an additional set of intermediate reference frames to use Viam's [motion service](/services/motion/).
```
Comment (sguequierre) [inline docs/internals/kinematic-chain-config.md:11]:
```suggestion
Many components have complex kinematic chains and require an additional set of intermediate reference frames to use Viam's [Motion service](/services/motion/).
```

File: docs/services/frame-system/_index.md [hunk 15]
```diff
@@ -0,0 +1,173 @@
...
+It stores the required contextual information to use the position and orientation readings returned by some components.
+
+## Configuration
+
+To adjust the Frame System from its default configuration for a particular [component](/components), add the following to its configuration:
+
+{{< tabs name="Frame Configuration Instructions" >}}
+{{% tab name="Config Builder" %}}
+
+Navigate to the **CONFIG** tab on your robot's page in [the Viam app](https://app.viam.com), select the **Builder** mode, scroll to a component's card, and click **Add Frame**:
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/services/frame-system/_index.md:24]: @npentrel Tried to address on line 28 below, LMK what you think

File: docs/services/frame-system/_index.md (hunk 4 above)
Comment (sguequierre) [reply docs/services/frame-system/_index.md:113]: done, LMK if you have any suggestions

File: docs/internals/kinematic-chain-config.md
[the remark this answers is not in the document]
Comment (spieswl) [reply docs/internals/kinematic-chain-config.md:1]: Documentation on `ModelJSON` likely belongs with the `arm` component, although the useful application _could_ be extended to other components in the future, so "Internals" seems like a safe middle-ground to me.

File: docs/services/frame-system/component-on-static.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+Measure from that point to the base of the arm to get the `"translation"` coordinates.
+
+- Suppose the arm is offset from the corner by 100mm in the positive X direction, and 250mm in the negative Y direction.
+- Supply this `"translation"` when configuring the arm component's `"frame"` information.
+- Leave `"parent"` and `"orientation"` at their default values.
+
+{{< tabs name="Example Frame Configuration of Component attached to Static Surface" >}}
+{{% tab name="Config Builder" %}}
+
+Navigate to the **CONFIG** tab on your robot's page in [the Viam app](https://app.viam.com), select the **Builder** mode, scroll to `"myArm"`'s card, and click **Add Frame**:
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/services/frame-system/component-on-static.md]: Done, I think!

File: docs/services/frame-system/_index.md (hunk 5 above)
Comment (spieswl) [reply docs/services/frame-system/_index.md:150]: @biotinker and I are going to talk about this together during our morning 1:1, so we should have some updates for you shortly.

**Review by spieswl — APPROVED** [review]
LGTM!

File: docs/services/frame-system.md
Comment (npentrel) [inline docs/services/frame-system.md:1]: this file needs to be deleted completely

File: docs/services/frame-system/component-on-dynamic.md [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+Measure from that point to the base of the arm to get the `"translation"` of the arm.
+
+- Suppose the base of the arm is mounted to the gantry 100mm above the gantry's origin.
+- Supply this `"translation"` and specify the arm's `"parent"` reference frame as `"myGantry"`.
+- Leave all other frames' `"orientation"` and `"translation"` at their default values.
+
+Now, as the gantry moves its actuator, the Frame System will translate both the gantry and the arm's location according to that motion.
+
+{{< tabs name="Example Frame Configuration of Component attached to Dynamic Component" >}}
+{{% tab name="Config Builder" %}}
```
Comment (npentrel) [inline docs/services/frame-system/component-on-dynamic.md:24]: Do you need to add parent, translation and orientation here?

File: docs/services/frame-system/_index.md [hunk 18]
```diff
@@ -0,0 +1,178 @@
...
+{{% /alert %}}
+
+For more information about determining the appropriate values for these parameters, see these two examples:
+
+- [A component attached to a static surface](/services/frame-system/component-on-static)
+- [A component attached to another, dynamic, component](/services/frame-system/component-on-dynamic)
+
+## Building the Frame System
+
+Access a [topologically-sorted list](https://en.wikipedia.org/wiki/Topological_sorting) of the generated reference frames in the robot's logs at `--debug` level:
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/services/frame-system/_index.md:125]: done!

File: docs/services/frame-system/_index.md [hunk 19]
```diff
@@ -0,0 +1,178 @@
...
+
+- You can pass a detected object's frame information to the `supplemental_transforms` parameter in your calls to Viam's Motion Service's [`GetPose`](/services/motion/#getpose) method.
+- Functions of some services and components also take in a `WorldState` parameter, which includes a `transforms` property.
+- Both [`TransformPose`](#accessing-the-frame-system) and [`FrameSystemConfig`](#accessing-the-frame-system) have the option to take in these supplemental transforms.
+
+{{% alert title="TBD" color="note" %}}
+
+Additional uses for supplemental transforms are being explored by the Viam team.
+
+{{% /alert %}}
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/services/frame-system/_index.md:178]: removed!

File: docs/internals/kinematic-chain-config.md [hunk 20]
```diff
@@ -0,0 +1,316 @@
...
+            "min": -360
+        }
+    ]
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+{{% alert title="Caution" color="caution" %}}
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/internals/kinematic-chain-config.md:312]: yeah it doesn't, done!

File: docs/services/frame-system/component-on-static.md [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+- Supply this `"translation"` when configuring the arm component's `"frame"` information.
+- Leave `"parent"` and `"orientation"` at their default values.
+
+{{< tabs name="Example Frame Configuration of Component attached to Static Surface" >}}
+{{% tab name="Config Builder" %}}
+
+To complete the frame configuration for your robot following this example, navigate to the **CONFIG** tab on your robot's page in [the Viam app](https://app.viam.com), select the **Builder** mode, scroll to `"myArm"`'s card, and click **Add Frame**:
+
+![frame card example for this configuration](../img/frame_card_static.png)
+
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/services/frame-system/component-on-static.md:26]: done

File: docs/services/frame-system/_index.md (hunk 8 above)
Comment (sguequierre) [reply docs/services/frame-system/_index.md:146]: https://viam.atlassian.net/browse/DOCS-674?atlOrigin=eyJpIjoiOTUzNmU1ZmZlYjA0NDAwYWEzZmZiODBmMDk5OTg2YzgiLCJwIjoiaiJ9

File: docs/services/frame-system/component-on-static.md (hunk 21 above)
[the remark this answers is not in the document]
Comment (npentrel) [reply docs/services/frame-system/component-on-static.md:26]: hmm now I am wondering if the parameter tables should  be on the subpages too

**Review by npentrel — APPROVED** [review]
LGTM - I'd suggest still adding the parameter tables on the subpages too but otherwise good to go

**Pull request merged**